src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/XmlDataContract.cs

```diff
@@ -131,34 +131,57 @@ public override XmlDictionaryString TopLevelElementNamespace
             { _helper.TopLevelElementNamespace = value; }
         }
 
-#if !NET_NATIVE
+#if NET_NATIVE
+        private CreateXmlSerializableDelegate _createXmlSerializableDelegate;
+        public CreateXmlSerializableDelegate CreateXmlSerializableDelegate        
+#else
         internal CreateXmlSerializableDelegate CreateXmlSerializableDelegate
+#endif
         {
+#if !NET_NATIVE
             /// <SecurityNote>
             /// Critical - fetches the critical CreateXmlSerializableDelegate property
             /// Safe - CreateXmlSerializableDelegate only needs to be protected for write; initialized in getter if null
             /// </SecurityNote>
             [SecuritySafeCritical]
             get
             {
-                if (_helper.CreateXmlSerializableDelegate == null)
+                if (DataContractSerializer.Option == SerializationOption.CodeGenOnly || DataContractSerializer.Option == SerializationOption.ReflectionAsBackup)
                 {
```

Comment:
Could you explain (maybe as a comment in the code) why this 'if' test was added?  Before, it executed the code unconditionally (on non-NET_NATIVE), and now it executes only under certain conditions.  It would help developers understand what is happening.  Not blocking.